packages/nx/src/tasks-runner/task-graph-utils.ts

```diff
@@ -35,6 +39,29 @@ export function findCycle(graph: {
   return null;
 }
 
+/**
+ * This function finds all cycles in the graph.
+ * @returns a list of unique task ids in all cycles found, or null if no cycle is found.
+ */
+export function findCycles(graph: {
+  dependencies: Record<string, string[]>;
+}): Set<string> | null {
+  const visited = {};
+  const cycles = [];
+  for (const t of Object.keys(graph.dependencies)) {
+    visited[t] = false;
+  }
+
+  for (const t of Object.keys(graph.dependencies)) {
+    const cycle = _findCycle(graph, t, visited, [t]);
+    if (cycle) {
+      cycles.push(cycle);
```

Comment:
I think it's better to just add to the set and avoid having to create the array and call `.flat()`.


```
cont cycles = new Set<string>();

// ...

  for (const t of Object.keys(graph.dependencies)) {
    const cycle = _findCycle(graph, t, visited, [t]);
    if (cycle) {
     cycles.add(cycle)
    }
}

// ...
```

And then return the set, I don't think you need to return `null`?

Reply:
changed